src/Tests/ImageBundleTest.php

```diff
@@ -0,0 +1,136 @@
+   * Tests image media bundle creation from config files.
+   */
+  public function testMediaBundleCreationFromModule() {
+    $type_configuration = array(
+      'source_field' => 'field_image',
+      'gather_exif' => false
+    );
+
+    $this->assertTrue((bool) $this->testBundle, 'The media bundle from default configuration has been created in the database.');
+    $this->assertEqual($this->testBundle->get('label'), 'Image', 'Correct label detected.');
+    $this->assertEqual($this->testBundle->get('description'), 'Use Image for uploading locally hosted images.', 'Correct description detected.');
+    $this->assertEqual($this->testBundle->get('type'), 'image', 'Correct plugin ID detected.');
+    $this->assertEqual($this->testBundle->get('type_configuration'), $type_configuration, 'Type configuration correct.');
+    $this->assertEqual($this->testBundle->get('field_map'), [], 'Correct field map detected.');
+  }
+
+  /**
+   * Tests item creation and thumbnail
+   */
+  public function testMediaBundleItemCreation() {
+    // Define the media item name.
+    $name = $this->randomMachineName();
+    $image_files = $this->drupalGetTestFiles('image');
+    $testImage = File::create((array) current($image_files));
+    $file_path = $this->container->get('file_system')->realpath($testImage->getFileUri());
+    $edit = [
+      'name[0][value]' => $name,
+      'files[field_image_0]' => $file_path,
+    ];
+
+    // Save the image
+    $this->drupalPostForm('media/add/' . $this->testBundle->id(), $edit, t('Save and publish'));
+    $this->drupalPostForm(NULL, ['field_image[0][alt]' => $name], t('Save and publish'));
+    // Verify the title
+    $this->assertTitle($edit['name[0][value]'] . ' | Drupal');
+
+    // Let's retrieve the media id and corresponding media entity object
+    $media_id = $this->container->get('entity.query')->get('media')->execute();
+    $media_id = reset($media_id);
+    /** @var \Drupal\media_entity\MediaInterface $media */
+    $media = $this->container->get('entity_type.manager')
+        ->getStorage('media')
+        ->loadUnchanged($media_id);
+
+    // Now let's grab the image, thumbnail and the thumbnail URL
+    $field_name = 'field_image';
+    $file = $media->{$field_name}->entity;
+    $image_uri = $file->getFileUri();
+    $thumbnail_url = ImageStyle::load('thumbnail')->buildUrl($image_uri);
+    // Extract itok from url
+    $parts = parse_url($thumbnail_url);
+    parse_str($parts['query'], $query);
+    $itok = $query['itok'];
+    $thumbnail_uri = ImageStyle::load('thumbnail')->buildUri($image_uri);
+    $src = file_url_transform_relative(file_create_url($thumbnail_uri));
+    $src = $src . '?itok=' . $itok;
+    $elements = $this->xpath(
+        '//img[@src=:url and @alt=:alt and @title=:title]',
+        array(
+            ':url' => $src,
+            ':title' => $name,
+            ':alt' => 'Thumbnail',
+        )
+    );
+    $this->assertEqual(count($elements), 1, 'Thumbnail found on the page.');
```

Comment:
I think you're overcomplicating this and still you're not actually checking if the uploaded image was converted into a thumbnail. This could still pass if the generic thumbnail was generated, wouldn't it?